client/src/actions/user.ts

```diff
@@ -14,6 +14,16 @@ export interface LoginActionFailure {
     error: string;
 }
 
+export interface RegisterFailure {
+    type: 'register_failure';
+    errors: string[];
+}
+
+export interface RegisterSuccess {
+    type: 'register_success';
+    next: string;
```

Comment:
`next` is kinda vague. Could we either throw down a comment or find a more descriptive name? I'm sure it'll eventually make sense but it's nice when typedefs are descriptive